connectors/src/lib/temporal.ts

```diff
@@ -65,3 +73,44 @@ export async function getTemporalWorkerConnection(): Promise<{
   const connection = await NativeConnection.connect(connectionOptions);
   return { connection, namespace: process.env.TEMPORAL_NAMESPACE };
 }
+
+export async function getConnectorId(
+  workflowRunId: string
+): Promise<ModelId | null> {
+  if (!WORKFLOW_ID2CONNECTOR_ID_CACHE[workflowRunId]) {
+    const client = await getTemporalClient();
+    const workflowHandle = await client.workflow.getHandle(workflowRunId);
+    const described = await workflowHandle.describe();
+    if (described.memo && described.memo.connectorId) {
+      if (typeof described.memo.connectorId === "number") {
```

Comment:
Unfortunately we have connectorId being transported as a string and a number in many places in connectors. We should move toward number but since there is no way to type what's in memo, I think supporting both is convenient.

Reply:
Can't we force the one passed into memo to be one or the other ?

Reply:
Got it thanks